Fix phase2 pixel in GeometrySearch

ebrondol opened the pull request:
This PR has been developed in CMSSW_8_1_X_2016-11-01-2300 and tested on top of the PR #16407. This is the evolution of the story: with a private branch that produce the hit on track maps for phase2 geometry, I produced the ratio clusters/clusters on track in D4:
![ontrackmultmap_v4021_baseline](https://cloud.githubusercontent.com/assets/5331004/20005936/65407394-a295-11e6-98c2-cb284fb1922e.png)
It is pretty clear that the forward phase2 pixel detector is missing some hits and the last rings of the last forward layers[8-11] are not included in the search. 
The problem has been cured using not anymore the class `PixelForwardLayerBuilder` but instead adapting `Phase2OTEndcapLayerBuilder` to be also used for the forward phase2 pixel. In this respect the naming has been changed everywhere.
After this fix, the new map for D4 looks like:
![ontrackmultmap_v4021_fixphase2pixelgeometry_rebase](https://cloud.githubusercontent.com/assets/5331004/20006102/56e5273a-a296-11e6-8257-0ef19732a0bc.png)
The code has been changed in such a way that the other D* should not be affected.

This impact some track variables such as the #hits vs eta, just two examples for [single mu](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161103_FourMuExtended/plots_ootb/hitsAndPt.pdf) and for [ttbar](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161103_TTbar/plots_ootb/hitsAndPt.pdf). But both seem now much more reasonable respect to before. If you want to have a look at the full set of results you can find them [here](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161103_FourMuExtended/) and [there](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161103_TTbar/). No huge differences have been noticed in eff/fakes.

The documentation in `RecoTracker/TkDetLayers/doc/TkDetLayersTree.pdf` has also been update - including the tilted update that was never introduced.

Thanks to @VinInn @rovere @venturia 
Informing @delaere @boudoul  @atricomi @kpedro88

ebrondol committed:
try but crash

ebrondol committed:
first attempt to include new phase2 pixel in the navigation

ebrondol committed:
Introducing Phase2PixelEndcapLayer

Temporary solution to partly fix the Navigation.
The problem is the "subDetector" method which should differciate the
Pixel Layers with OT ones. This is a dirty solution, but it is working.
Better to think about templating the class..

ebrondol committed:
cleaning up

ebrondol committed:
fix bug - OT rings always with Brothers!

ebrondol committed:
Fix Navigation School for phase2 pixel

The last ring of the new layout for phase2 pixel extend in r over the
min r of the OT. This brings a bug in the navigation that has been
fixed.

ebrondol committed:
fix ordering scheme in forward geometry

ebrondol committed:
fix small bug

ebrondol committed:
fix other merging problems

ebrondol committed:
Removing Phase2PixelEndcap and changing names

The Phase2PixelEndcap is not needed anymore, only the classes
Phase2EndcapLayer and Phase2EndcapRing are used for both Pixel and OT
detector. The names have been changed accordingly.

ebrondol committed:
clarifying comments

ebrondol committed:
fix compatibility with D1

ebrondol committed:
from at() to []

ebrondol committed:
updating phase2 part of TkDetLayersTree.pdf

cmsbuild commented:
A new Pull Request was created by  @ebrondol  for CMSSW_8_1_X.

It involves the following packages:

RecoTracker/TkDetLayers
RecoTracker/TkNavigation
RecoTracker/TkSeedGenerator


@cmsbuild, @cvuosalo, @slava77, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @makortel, @GiacomoSguazzoni, @rovere, @VinInn, @mschrode, @gpetruc, @dgulhan this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
@ebrondol 
which workflow did you use to make "ratio clusters/clusters on track in D4" ?
I'd expect a muon gun or something similarly dominated by "real and trackable" particles is used.

ebrondol commented:
@slava77 yes, in this particular plot I have used the 4 muons wf with pt between 1 and 200 GeV (extended). I reproduced the same also with single mu pt 10 GeV.

ebrondol commented:
ah, I worked on top of the PR #16407, so I had all of them automatically.

cmsbuild commented:
+1
Tested at: 54052f20680500fe58ab9e56be2d47a35a706dd6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16470/16203/summary.html

cmsbuild commented:
Comparison job queued.

kpedro88 commented:
@boudoul I have a question... maybe it is just my ignorance of the Phase2 tracker design, but there seems to be an asymmetry for |z| < ~100 and r < 20. For z < 0 there are 6 vertical lines, but for z > 0 there are 7 vertical lines. Is this intended?

boudoul commented:
Errr.... certainly not naive and certainly not expected... looks like disc2 on minus side disappeared .... (?)

venturia commented:
On 04 Nov 2016, at 16:10 , boudoul <notifications@github.com<mailto:notifications@github.com>> wrote:


Errr.... certainly not naive and certainly not expected... looks like disc2 on minus side disappeared .... (?)


It could be simply a problem in the configuration of Erica’s analysis job. Each module in that plot is a line in the configuration file, eventually.


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/16470#issuecomment-258457331>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEx-p52lOGRRgh-RiOCQlnfSAEvK3nsAks5q60rOgaJpZM4Kphg4>.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16470/16203/summary.html

ebrondol commented:
@kpedro88 thanks for noticing it. I will come back asap with an explanation.

ebrondol commented:
@kpedro88 @boudoul 
Yes, as @venturia said, it was just an error in my analyzer. The track map is :
![ontrackmultmap_v4021_fixphase2pixelgeometry_rebase_v2](https://cloud.githubusercontent.com/assets/5331004/20059269/a2fb7636-a4f5-11e6-8fd6-d170464f0f8e.png)

ebrondol commented:
@slava77 About the following issue: "D4 tracking layout increases the number of hits and also fake rate a bit: this adds maybe 1/4 to the CPU time. " should be solved with this PR - at least what the results that I have produced locally. Please test it and let me know! Thanks a lot!

venturia commented:
@ebrondol now you may be missing one module selection in the first layer of the barrel pixel detector as pointed out at today upgrade performance meeting: there is a missing module in your Tracker cross-section plot. Not a big deal for this PR

ebrondol commented:
@venturia yes, I correct it again. Thanks for noticing it.

slava77 commented:
@ebrondol 
if there is something to fix here relevant to this update, it might be better to do it now instead of delaying to another PR.

ebrondol commented:
@slava77 I do agree, but the pixel shape cluster filter does not depend on me.
@tricomi @delaere @boudoul

slava77 commented on RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc:
```diff
@@ -0,0 +1,100 @@
+        frontGeomDets.push_back(theGeomDet);
+  
+      if( fabs( (*compGeometricDets)->positionBounds().z() ) > fabs(meanZ))
+        backGeomDets.push_back(theGeomDet);      
```
is equality forbidden or is it to be ignored?
For the former case, maybe there should be an assert or an exception.

ebrondol replied:
It should never be the case, since the mean is computed from the `compGeometricDets` themselves. But we can put an Exception to be safe in any case.

slava77 replied:
ok, looks like no change necessary

slava77 commented on RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:
```diff
@@ -126,6 +126,13 @@ void SimpleNavigationSchool::linkNextForwardLayer( const BarrelDetLayer* bl,
 	 radius < (**fli).specificSurface().outerRadius()) {
+      //search if there are any sovrapposition between forward layers
+      for ( FDLI fliNext = fli; fliNext != theRightLayers.end(); fliNext++) {
+        if ( (**fliNext).position().z() < (**fli).position().z() && (**fliNext).specificSurface().innerRadius() < (**fli).specificSurface().outerRadius()) {
```
is Z always positive in this context?

ebrondol replied:
Yes, it is because they are `theRightLayers` which means Forward Layers with z > 0
But maybe we can put en exception also in this case, if you prefer.

ebrondol replied:
Not sure if I should introduce the Exception for both `fli` and `fliNext` or just `fli` (since this is a loop on the layers).. @slava77 what do you suggest?

ebrondol committed:
introducing exception and PixelShapeFile for phase 2

cmsbuild commented:
Pull request #16470 was updated. @cmsbuild, @cvuosalo, @slava77, @davidlange6 can you please check and sign again.

kpedro88 commented:
@ebrondol is this pixel shape file going to fix the CPU usage?

ebrondol commented:
@slava77 I introduced the exception in `RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc` because I think it is safer in this way, please have a look. Also I introduced the new cuts on the pixel shape produced by Konstantin Androsov (not sure how to mention him) and I am producing final checks.

@atricomi @boudoul @delaere

ebrondol commented:
@kpedro88 hopefully this will make some effect, I did not try by myself yet. I would like to check the performance first.

venturia commented:
In my opinion if the pixel cluster shape filter is considered so critical on the performance, both positively and negatively, it would be better to monitor more closely the simplest observable which is affected by this filter, namely, if I am not mistaken, the number of triplets (or of seeds made by triplets) as a function of your preferred quantity (pileup, number of pixel clusters, number of gentracks,...)

slava77 commented:
@cmsbuild please test

@ebrondol 
the last change looks OK visually

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 73649c382ef2fb1781d6f4838099812b3c9395fb
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16470/16390/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16470/16390/summary.html

ebrondol commented:
@slava77 here the performance on [single mu](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161116_FourMuExtended_D4/) and [ttbar](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161116_TTbar/) with D4. They seem really good to me.

ebrondol commented:
@venturia I think that the main changes on performance of the filter is when it is wrong. This is why I wanted to double check that the performance were as expected. When it is off (as in the case of the baseline) or when it has the right correction (as in the case of this PR), this does not seem to affect the tracking performance - as far as I can tell. The problem we found out in the case of filter off, stands in the CPU timing.
@slava77 please, can you test now the timing situation with this PR in?

slava77 commented:
On 11/17/16 1:17 AM, ebrondol wrote:
> @slava77 <https://github.com/slava77> here the performance on single mu
> <http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161116_FourMuExtended_D4/>
> and ttbar
> <http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161116_TTbar/>
> with D4.

This shows almost no effect on eff or fakes
which ttbar is this? (21224 in the plots suggests no PU)
The bad clusters may have been from PU and OOT.

> They seem really good to me.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16470#issuecomment-261194859>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbihQQpO3CehwCDimLUIZPwfI_e50ks5q_BuygaJpZM4Kphg4>.
>

slava77 commented on RecoPixelVertexing/PixelLowPtUtilities/python/ClusterShapeHitFilterESProducer_cfi.py:
```diff
@@ -11,6 +11,5 @@
 phase2_tracker.toModify(ClusterShapeHitFilterESProducer,
-    PixelShapeFile = 'RecoPixelVertexing/PixelLowPtUtilities/data/pixelShape_Phase1Tk.par',
-    doPixelShapeCut = cms.bool(False)
+    PixelShapeFile = 'RecoPixelVertexing/PixelLowPtUtilities/data/pixelShape_Phase2Tk.par',
```
is this applicable to T1,T2 geometries?
I thought that the pixelShape_Phase1Tk.par was supposed to be applied to T1,T2 (phase-1 like pixel).
Most of the workflows that we have are still T1. It would be bad if behavior there goes bad with pixelShape_Phase2Tk.

ebrondol replied:
You are right. 
I am not sure how to menage this problem in the different geometries.
@kpedro88 @atricomi @boudoul do you have any suggestion?

makortel replied:
I think the simplest is to have T1+T2 have different tracker-era than T3.

makortel replied:
Let me remind that the most correct solution would be to move the whole calibration to GT (which would have saved some hairs with phase1 in pre12), but probably we are, again, after a "quicker" solution.

ebrondol replied:
What if we check before the performance of T1 and T2 with this "wrong" calibration? In any case also the "phase1" ones were not optimal. I know that they are not the right ones for them, but if T3 is the only that we care for TDRs, and the performance on T1 and T2 are also kind of ok with the phase2Par, we do not need to create other eras or so which takes time and maybe confusion later.
I have started to produce results for [ttbar](http://ebrondol.web.cern.ch/ebrondol/Phase2Tracking/fixPhase2PixelGeometry/20161116_TTbar_D1/plots_ootb/effandfake1.pdf) in D1 geometry, with and without this PR. It does seem to me evetything all right.. @slava77 what do you think?

ebrondol replied:
And the result basically means: no so bad if we use phase2 shape filter for phase1, while a disaster if we do the contrary.. interesting.

kpedro88 replied:
The addition of multiple phase2_tracker Eras is possible, but introduces a lot of complexity. If @ebrondol can show the performance is not too bad, it would be easier just to use the same pixel shape everywhere. (@makortel: moving to GT would imply we'd have to create a separate GT queue for each Phase2 detector version... also not ideal.)

slava77 replied:
what are the results with PU?

Unless we move all "standard" workflows to use T3 (to include variations with I1/I2, M1/M2, timing or no timing), we should first confirm that these phase2 shapes are ok on T1 in the full dyn range (well, if it helps to speedup the tests, PU140 may be enough)

slava77 replied:
as I pointed in a separate email, can we look at comparisons of selections in phase1 and phase2 files compared to how the clusters look like (for good and bad clusters)?
So far the discussion treats the file and overall tracking as black boxes and we just look at high level outputs.

slava77 commented:
> @slava77 <https://github.com/slava77> please, can you test now the
> timing situation with this PR in?

I'll start the job soon.
For now, I have numbers with the older version of this PR (just improving the geom search):
they are roughly unchanged.

```
  delta/mean    delta/meanJob   baseline               new
   -0.724155      -0.70%    143217.00 ms/ev ->     67075.00 ms/ev initialStepTrackCandidates
   -0.563821      -1.32%    327464.00 ms/ev ->    183436.00 ms/ev lowPtTripletStepTrackCandidates
   -0.259058      -0.18%     85091.60 ms/ev ->     65575.80 ms/ev electronCkfTrackCandidates
   -0.207175      -0.19%    111620.00 ms/ev ->     90665.70 ms/ev lowPtQuadStepTrackCandidates
   +0.178413      +0.66%    366245.00 ms/ev ->    437988.00 ms/ev detachedQuadStepSeeds
   +0.071200      +1.46%   2154320.00 ms/ev ->   2313370.00 ms/ev allConversions
   +0.067070      +1.69%   2656590.00 ms/ev ->   2840950.00 ms/ev ecalDrivenElectronSeeds
   +0.063588      +0.24%    398012.00 ms/ev ->    424152.00 ms/ev lowPtQuadStepSeeds
```
there were actually improvements in a fraction of iterations, but these were balanced out by increases in some others, and most notably in allConversions and ecalDrivenElectronSeeds

ebrondol commented:
@slava77 yes, this is ttbar from 21224 with no PU.

atricomi commented:
We should have different files for T1-T2 and T3. IS it possible to have different eras?


Il giorno 17/nov/2016, alle ore 15:05, ebrondol <notifications@github.com> ha scritto:

> @ebrondol commented on this pull request.
> 
> In RecoPixelVertexing/PixelLowPtUtilities/python/ClusterShapeHitFilterESProducer_cfi.py:
> 
> > @@ -11,6 +11,5 @@
>  )
>  from Configuration.Eras.Modifier_phase2_tracker_cff import phase2_tracker
>  phase2_tracker.toModify(ClusterShapeHitFilterESProducer,
> -    PixelShapeFile = 'RecoPixelVertexing/PixelLowPtUtilities/data/pixelShape_Phase1Tk.par',
> -    doPixelShapeCut = cms.bool(False)
> +    PixelShapeFile = 'RecoPixelVertexing/PixelLowPtUtilities/data/pixelShape_Phase2Tk.par',
> You are right. 
> I am not sure how to menage this problem in the different geometries.
> @kpedro88 @atricomi @boudoul do you have any suggestion?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

slava77 commented:
Some plots with the older version (as of 54052f2 ), these should show the effect of the geometry search alone

ttbar PU200 (derived from 21224)
efficiency and fake rate are roughly similar and dup rate is down
<img width="855" alt="wf21224pu_dup_vs_eta" src="https://cloud.githubusercontent.com/assets/4676718/20403478/ca8d8384-acb5-11e6-9dcc-b462470b0223.png">
<img width="871" alt="wf21224pu_eff_vs_eta" src="https://cloud.githubusercontent.com/assets/4676718/20403482/ce1efce4-acb5-11e6-9253-82b2d7585c08.png">
<img width="862" alt="wf21224pu_fr_vs_eta" src="https://cloud.githubusercontent.com/assets/4676718/20403485/d0c10cee-acb5-11e6-8c41-26a384e2bd2a.png">
(roughly the same picture without pt0.9 cut)

the largest change in tracking steps appears to be in pixelPair building
<img width="854" alt="wf21224pu_pixelpaircand_eta" src="https://cloud.githubusercontent.com/assets/4676718/20403513/ef5a6830-acb5-11e6-800a-a383f183364a.png">

chi2 and nhits distribution look in agreement with plots posted with the PR itself.

this should be useful for incremental comparison. Since the pixel shape selection is a more significant feature.

slava77 commented:
is this PR needed in 81X? we can just continue in 90X.

the fixes are rather minor in terms of the impact (jobs run, efficiency and fake rate is about the same; timing is not reduced by much, just under 20% in D4 PU200 tests).
So, it doesn't really qualify as a significant bug fix in 81X.


we can continue discussion in this thread though for simplicity

ebrondol commented:
H @slava77 , I am not sure which release will be used for the final production of the TDR. This PR must be for sure in that release. 
For validation purposes, I would like to have it also in 81X, because this PR fix the distribution of number of hits and the shape of the clusters. Maybe this effect can be important for other physics studies (such as b-tagging).

slava77 commented:
On 11/20/16 5:37 AM, ebrondol wrote:
> H @slava77 <https://github.com/slava77> , I am not sure which release
> will be used for the final production of the TDR. This PR must be for
> sure in that release.

It will be based on 90X.


> For validation purposes, I would like to have it also in 81X, because
> this PR fix the distribution of number of hits and the shape of the
> clusters. Maybe this effect can be important for other physics studies
> (such as b-tagging).
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16470#issuecomment-261778841>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbocovkin5cq0trGRCUpeTFmxkeBgks5rAE0egaJpZM4Kphg4>.
>

slava77 commented:
Here are some notes on the impact of the last changes (https://github.com/cms-sw/cmssw/compare/54052f2...73649c3 which is due to the pixel shape selections).

Based on tests in CMSSW_8_1_X_2016-11-15-2300 for D4 ttbar with PU200.

I was looking for low-level plots like OffTrack or OnTrack, but apparently they are not in the phase-2 DQM. These would've been useful.

The main impact is on the technical side: CPU decreases by about 15% for the job and by a much more significant fraction for the affected modules (comparing to the IB here):
```
delta/moduleMean  delta/jobMean baseline       PR
   -1.883030      -7.50%    845426.00 ms/ev ->     25467.00 ms/ev pixelTracks
   -0.880414      -2.23%    398012.00 ms/ev ->    154703.00 ms/ev lowPtQuadStepSeeds
   -0.699192      -0.16%     32973.60 ms/ev ->     15890.80 ms/ev lowPtTripletStepSeeds
   -0.695042      -0.68%    143217.00 ms/ev ->     69346.70 ms/ev initialStepTrackCandidates
   -0.501832      -0.07%     18471.40 ms/ev ->     11061.20 ms/ev highPtTripletStepSeeds
   -0.357196      -0.91%    327464.00 ms/ev ->    228220.00 ms/ev lowPtTripletStepTrackCandidates
   +0.293221      +0.11%     33786.20 ms/ev ->     45395.00 ms/ev unsortedOfflinePrimaryVertices
   -0.284970      -0.08%     37125.10 ms/ev ->     27865.00 ms/ev detachedQuadStepTrackCandidates
   -0.247212      -0.17%     85091.60 ms/ev ->     66370.00 ms/ev electronCkfTrackCandidates
   -0.245667      -0.22%    111620.00 ms/ev ->     87198.40 ms/ev lowPtQuadStepTrackCandidates
   -0.215139      -0.65%    366245.00 ms/ev ->    295104.00 ms/ev detachedQuadStepSeeds
   -0.213871      -0.09%     49545.00 ms/ev ->     39972.40 ms/ev initialStepSeeds

```
Note that despite a factor of 30 change in CPU in pixelTracks, the final number of pixelTracks reconstructed is about the same.

Incremental differences of tracking performance (54052f2..73649c3) show fairly small impact on high level values (eff and fake rate)
<img width="859" alt="wf21224pu_73649c3vs54052f2_general_dup_v_eta" src="https://cloud.githubusercontent.com/assets/4676718/20465790/f39267ce-af1a-11e6-9a8b-01a19f151a56.png">
<img width="850" alt="wf21224pu_73649c3vs54052f2_general_dup_v_pt" src="https://cloud.githubusercontent.com/assets/4676718/20465791/f5d8449a-af1a-11e6-92de-17b005887a1a.png">
<img width="850" alt="wf21224pu_73649c3vs54052f2_general_eff_v_pt" src="https://cloud.githubusercontent.com/assets/4676718/20465792/fb4fb8e0-af1a-11e6-8aa3-953ed8fd9651.png">
<img width="857" alt="wf21224pu_73649c3vs54052f2_general_fr_v_eta" src="https://cloud.githubusercontent.com/assets/4676718/20465796/01f0cf2c-af1b-11e6-9154-539968214db4.png">

as expected, there is a slight decrease in the number of hits
<img width="853" alt="wf21224pu_73649c3vs54052f2_general_pxb_v_eta" src="https://cloud.githubusercontent.com/assets/4676718/20465798/1147bfc6-af1b-11e6-8801-7f2cf858187b.png">

there is some reshuffle in the final track algos ( more tracks in lowPtTriplet)
<img width="871" alt="wf21224pu_73649c3vs54052f2_general_algo" src="https://cloud.githubusercontent.com/assets/4676718/20465806/2a623dba-af1b-11e6-9967-4b24c1310280.png">

lowPtTriplet has less seeds
<img width="873" alt="wf21224pu_73649c3vs54052f2_seedeta_lowptt" src="https://cloud.githubusercontent.com/assets/4676718/20465809/3bc604a6-af1b-11e6-812a-c5c81ea89188.png">
but the number of candidates is much larger
<img width="871" alt="wf21224pu_73649c3vs54052f2_candeta_lowptt" src="https://cloud.githubusercontent.com/assets/4676718/20465816/73eaa152-af1b-11e6-9641-018e5525d3a9.png">

Some inclusive statistics: 
- 10% more tracks total (same for pf ch), concentrated in pt<0.7 GeV
- ~20% more offlinePVs
- higher PT objects are not really affected

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/16470 73649c3
- changes are in line with the description and the follow up update to recover pixel shape selections. Even though the selections are relevant for T3 pixels, the no-pu tests show that these selections work well also for T1 for the signal tracks.
- jenkins tests pass and comparisons with baseline show differences only in 2023 workflows
- local tests confirm the expected behavior: significant decrease in CPU thanks to the reintroduced pixel shape selections, fairly small effect on fakes and efficiencies; flattening of chi2 and the change in the number of hits (these latter features are from the original PR submission, see more plots in the description).

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

davidlange6 commented:
+1
